src/domain/core/src/use_cases/create_dataset_use_case.rs

```diff
@@ -19,7 +19,24 @@ pub trait CreateDatasetUseCase: Send + Sync {
         &self,
         dataset_alias: &DatasetAlias,
         seed_block: MetadataBlockTyped<Seed>,
+        options: CreateDatasetUseCaseOptions,
     ) -> Result<CreateDatasetResult, CreateDatasetError>;
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone)]
+pub struct CreateDatasetUseCaseOptions {
+    pub dataset_visibility: DatasetVisibility,
+}
+
+// Used primarily for tests
+impl Default for CreateDatasetUseCaseOptions {
+    fn default() -> Self {
+        Self {
+            dataset_visibility: DatasetVisibility::PubliclyAvailable,
```

Comment:
I don't think that convenience of writing tests should dictate what defaults we use.

Our defaults should either:
- favor security by default
- be non-opinionated (i.e. `visibility: None`) letting the service itself decide what default visibility is

The latter option would let us provides the defaults on DI / config level when constructing the service.

Reply:
> favor security by default

I chose this way because it seemed to me more suitable in the case of use case parameters

--

Fixed